datatypes/src/primitives/coordinate.rs

```diff
@@ -8,20 +8,39 @@ use ocl::OclPrm;
 use postgres_types::{FromSql, ToSql};
 use proj::Coord;
 use serde::{Deserialize, Serialize};
+use std::hash::{Hash, Hasher};
 use std::{
     fmt,
     ops::{Add, Div, Mul, Sub},
     slice,
 };
 
-#[derive(Clone, Copy, Debug, Deserialize, PartialEq, PartialOrd, Serialize, Default)]
+#[derive(Clone, Copy, Debug, Deserialize, PartialOrd, Serialize, Default)]
 #[cfg_attr(feature = "postgres", derive(ToSql, FromSql))]
 #[repr(C)]
 pub struct Coordinate2D {
     pub x: f64,
     pub y: f64,
 }
 
+impl PartialEq for Coordinate2D {
+    fn eq(&self, other: &Self) -> bool {
+        crate::util::helpers::f64_to_bits(self.x) == crate::util::helpers::f64_to_bits(other.x)
+            && crate::util::helpers::f64_to_bits(self.y)
+                == crate::util::helpers::f64_to_bits(other.y)
+    }
+}
```

Comment:
Why do we need a special f64 comparison?

Reply:
377a1adcaf72e31a8346ad2a681033d2d21dfc02